### ISSUE TITLE
ExpansionPanel with no elevation and Colors.transparent has grey overlay

### ISSUE THREAD

**Issue by TheJulianJES** [flutter/flutter#89490]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows:
<details>
  <summary>Code</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  bool isExpanded1 = false;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(title: const Text('Test')),
        backgroundColor: Colors.green,
        body: ListView(
          children: [
            ExpansionPanelList(
              animationDuration: const Duration(milliseconds: 400),
              elevation: 0,
              expansionCallback: (int panelIndex, bool isExpanded) => setState(() => isExpanded1 = !isExpanded1),
              expandedHeaderPadding: EdgeInsets.zero,
              children: <ExpansionPanel>[
                ExpansionPanel(
                  backgroundColor: Colors.transparent,
                  canTapOnHeader: true,
                  headerBuilder: (BuildContext context, bool isExpanded) => Column(
                    children: const <Widget>[
                      ListTile(
                        title: Text('Title 3'),
                        subtitle: Text('Subtitle'),
                        contentPadding: EdgeInsets.symmetric(horizontal: 12.0),
                      ),
                    ],
                  ),
                  isExpanded: isExpanded1,
                  body: const SizedBox(),
                ),
              ],
            )
          ],
        ),
      ),
    );
  }
}
```
</details>

3. Launch on Xcode iOS simulator
4. Launch on a physical iOS device
5. Observe that there's a different "output" (real iOS device has a grey "overlay") 

**Expected results:** 
I expected the ExpansionPanel to be transparent if I provide Colors.transparent.

**Actual results:** 
On a real iOS device (tested on multiple iOS versions), I get a grey color on the ExpansionPanel:
<img src="https://user-images.githubusercontent.com/6409465/132120375-20268f3e-3b4a-4bde-9b31-157dc21b1499.PNG" width=25% height=25%>

However, with the iOS simulator, it works as expected:
<img src="https://user-images.githubusercontent.com/6409465/132120393-75cc2d32-ab25-4cbc-915e-ff553a82cb86.png" width=25% height=25%>

**Additional notes:**
I remember this being an issue ~~(maybe even pre-Flutter 2.0?)~~.
I tried the same thing again on another project today and the same issue occurred again.
However, there was a Flutter version where the grey overlay wasn't there on a real iOS device.
It seems like this bug has existed for quite some time though and I can't remember the last version where the bug wasn't present. It still seems to be a regression.
Web (dartped.dev) also doesn't show a grey overlay. I haven't tested it on Android but I'd assume that it also doesn't show a grey overlay.

<details>
  <summary>Logs</summary>

```console
Analyzing testj...                                                      
No issues found! (ran in 2.0s)
```

```console
[✓] Flutter (Channel beta, 2.5.0-5.3.pre, on macOS 12.0 21A5506j darwin-x64, locale de-DE)
    • Flutter version 2.5.0-5.3.pre at /Users/username/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a7fb06d6fa (3 days ago), 2021-09-02 09:07:14 -0700
    • Engine revision fcde2c8b2f
    • Dart version 2.14.0 (build 2.14.0-377.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/username/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 13.0, Build version 13A5212g
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (4 available)
    • iPhone 12 Pro (mobile) • xxxxxxxx            • ios            • iOS 15.0 19A5340a
    • iPhone 12 Pro Max (mobile)        • xxxxxxxx • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)                   • macos                                • darwin-x64     • macOS 12.0 21A5506j darwin-x64
    • Chrome (web)                      • chrome                               • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @TheJulianJES, thanks for filing the issue. 

I was unable to reproduce it with `stable` on my `iPhone xs`, however, I was able to reproduce it on `master`. Oddly enough, the same behaviour exists on both my simulator and my iphone.

On `web`, using the `html` renderer, the expected behaviour occurs, however, using the `canvaskit` renderer, the grey overlay is visible.

<details>
<summary> screenshots</summary>

| device | stable | master | 
|--------| -------| -------|
| iphone xs| ![IMG_2903](https://user-images.githubusercontent.com/88313112/132196799-da6abe8d-9242-41c0-a90c-b054f587865d.PNG)| ![IMG_2902](https://user-images.githubusercontent.com/88313112/132196837-8c34a029-1b59-4b87-bafc-cc31b4248acb.PNG)| 
| poco x3 nfc|![photo_2021-09-06 09 40 33](https://user-images.githubusercontent.com/88313112/132197000-f0b2ef02-70b0-423c-b42b-3a0c0deb3b66.jpeg)|![photo_2021-09-06 09 40 35](https://user-images.githubusercontent.com/88313112/132197053-695a16ee-9cbb-4ef5-a6ee-20922d5dbdcb.jpeg)|
| chrome | <img width="976" alt="Screenshot 2021-09-06 at 08 51 00" src="https://user-images.githubusercontent.com/88313112/132197403-4e6de478-12e2-4c30-91a1-5ef5992481e9.png">| <img width="976" alt="Screenshot 2021-09-06 at 08 47 54" src="https://user-images.githubusercontent.com/88313112/132197435-12d11743-c3bb-4913-9f42-cc41486818e8.png">|
| macOS | <img width="912" alt="Screenshot 2021-09-06 at 08 52 18" src="https://user-images.githubusercontent.com/88313112/132197539-6ffac19a-0b9b-410a-b91d-67b4bc2dd5c2.png">|<img width="912" alt="Screenshot 2021-09-06 at 08 55 00" src="https://user-images.githubusercontent.com/88313112/132197568-bf268f54-36e6-4f37-be33-576313d174de.png">|

</details>

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-arm, locale
    en-GB)
    • Flutter version 2.2.3 at /Users/nexus/dev/sdks/flutter
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (4 available)
    • M2007J20CG (mobile)    • 5dd3be00                  • android-arm64  •
      Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.7.1
    • macOS (desktop)        • macos                     • darwin-arm64   •
      macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 93.0.4577.63

• No issues found!
```


```console
[✓] Flutter (Channel master, 2.6.0-1.0.pre.193, on macOS 11.5.1 20G80
    darwin-arm, locale en-GB)
    • Flutter version 2.6.0-1.0.pre.193 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0c5431d99c (7 hours ago), 2021-09-05 22:31:02 -0400
    • Engine revision b9c633900e
    • Dart version 2.15.0 (build 2.15.0-82.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (4 available)
    • M2007J20CG (mobile)    • 5dd3be00                  • android-arm64  •
      Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.7.1 18G82
    • macOS (desktop)        • macos                     • darwin-arm64   •
      macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 93.0.4577.63

• No issues found!
```
</details>

**Comment by nt4f04uNd** [comment]
Android and Windows doesn't have that issue. Same for the Windows web (both renderers)

**Comment by TheJulianJES** [comment]
Also present on the stable channel with Flutter 2.5.0 now.
<details>
  <summary>Logs</summary>
[✓] Flutter (Channel stable, 2.5.0, on macOS 12.0 21A5506j darwin-x64, locale de-DE)
    • Flutter version 2.5.0 at /Users/xxx/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (13 hours ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/xxx/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 13.0, Build version 13A5212g
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • xxxxxxx • ios            • iOS 15.0 19A5340a
    • macOS (desktop)                   • macos                     • darwin-x64     • macOS 12.0 21A5506j darwin-x64
    • Chrome (web)                      • chrome                    • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
</details>

**Comment by jason-simmons** [comment]
Reproduced this on a Huawei Nova 3.

This is related to https://github.com/flutter/flutter/commit/27cc8c06b2a83b3ad6a4448b6e18045ddefc2d3c

The old implementation of `_RenderMergeableMaterialListBody._paintShadows` would only call `canvas.drawShadow` if `kElevationToShadow` defined shadows for the given elevation.  `kElevationToShadow[0]` was empty, so the `drawShadow` call would be skipped for the `elevation: 0` object in this example.

The new implementation is always calling `drawShadow`.  For some reason the call to Skia's `SkShadowUtils::DrawShadow` for elevation zero is acting inconsistently on various devices.

**Comment by TheJulianJES** [comment]
I know I've already mentioned this in the PR, but just wanted to mention it here again:

Is it possible to get the fix cherry-picked into the next 2.5.x version (or at least evaluate cherry-picking it)?
It was **working on 2.2.3 (previous stable** version) and is now **broken in 2.5.0 (current stable)**.

**Comment by UdaraWanasinghe** [comment]
Issue persists on iPhone 11. Are there any temporary fix for this?
